### ISSUE TITLE
unnecessary_lambdas should not trigger on extension type external methods

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#5015]
With the following code:

```dart
extension type MapType._(JSObject _) implements JSObject {
  @JS('f')
  external void _f();
  void Function() get f => () => _f();
}
```

a lint `unnecessary_lambdas` appears on `f` getter.

But if I use a tear-off:

```dart
extension type MapType._(JSObject _) implements JSObject {
  @JS('f')
  external void _f();
  void Function() get f => _f;
}
```

then I get an error:

> Error: Tear-offs of external extension type interop member '_f' are disallowed.

So `unnecessary_lambdas` should not trigger on extension type external methods

(Dart 3.4.4)

**Comment by lrhn** [comment]
Or maybe interop should support tear-of by generating a `() => this._f()` Dart function that you can tear of instead of the native function.
